comp/dogstatsd/server/server.go

```diff
@@ -605,6 +609,12 @@ func (s *server) handleMessages() {
 }
 
 func (s *server) restoreBlocklistFromLocalConfig() {
+	s.log.Debug("Restoring blocklist with local config.")
+
+	// update the runtime config to be consistent
+	// in `agent config` calls.
+	s.config.Set("statsd_metric_blocklist", s.localBlocklistConfig.metricNames, model.SourceAgentRuntime)
+
 	s.SetBlocklist(
```

Comment:
I think this `SetBlocklist` call will immediately set the config source back to `model.SourceRC` no?